_drafts/GDB Tips and Tricks.md

```diff
@@ -0,0 +1,500 @@
+#CFLAGS += $(OPT)
+CFLAGS += -ggdb
+```
+
+We commented out the optimization flags and added the `-ggdb` option which includes all debugging symbols needed for GDB.
+
+Now, we can go ahead and build the project. To do this, run the following command:
+
+```terminal
+$ make
+```
+
+The output should look something like this:
+
+
+![](Graphics/Make_Output.png)
+
+
+### 2. Hardware Setup
+The second step is to get the hardware set up to run the example.
+
+Here are the steps to accomplish this:
+
+- Connect the development kit to your computer (via the micro-USB to USB-A cable)
+- Make sure the development kit jumpers are in the right settings:
+	- **SW6** set to "Default" (**not** "nRF ONLY")
+	- **SW9** (nRF power source) set to "VDD"
+	- **Power** set to "ON"
+
+	![](Graphics/nRF52840_dev_kit.png)
+	
+- Erase the development kit by running the following command:
+
+	```terminal
+	$ nrfjprog -f NRF52 --eraseall
+	```
+	
+- The final step is to flash the development kit with the example's binary.
+
+	To do so, run the following command from the folder `<nRF5_SDK_Folder>/examples/peripheral/blinky/pca10056/blank/armgcc`.
+
+	```terminal
+	$ nrfjprog -f NRF52 --program _build/nrf52840_xxaa.hex --chiperase
+	```
+	
+That's it!
+
+Now the nRF52840 development kit should be flashed with the UART example and it should be running properly.
+
+To verify that, we will be using a serial terminal program.
+
+### 3. Serial Terminal Program Setup
+As we mentioned earlier, we'll be using a program called CoolTerm.
+
+There are three parts to get this working:
+
+- Run the CoolTerm application
+- Make sure the serial port settings are correct (listed at [this link](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.3.0/uart_example.html)):
+ 
+	
+	![](Graphics/CoolTerm_Settings.png)
+	
+	Now, hit **OK**. 
+
+- Finally, connect to the serial port by clicking the "Connect" button:
+	
+	![](Graphics/CoolTerm_Connect.png)
+	
+	You may not see any output since the program probably started before you connected. To reset the development board, we can simply run the following command from the Terminal:
+	
+	```terminal
+	$ nrfjprog -f NRF52 --reset
+	```
+	
+	If all goes well, you should see the following printed in the Terminal window:
+	
+	
+	![](Graphics/CoolTerm_Output.png)
+	
+
+## Debugging the Program
+Now that we have the application running properly, let's go ahead and set up the debugger.
+
+### 1. Remote GDB Setup
+There are a few steps to get this working.
+
+- **JLinkGDBServer**
+	
+	As part of the SEGGER J-Link Software we installed, the program JLinkGDBServer is included. This is the GDB Server application that will interface directly with the development kit and the nRF52840 chipset. It will open up a port (we chose port 2331) over the network to allow connections from a GDB Client.
+	
+	To start the J-Link GDB Server, run the following command:
+	
+	```terminal
+	$ JLinkGDBServerCL -device nrf52840_xxaa -if swd -port 2331
```

Comment:
would be nice to explain that `-if swd` selects a SWD interface rather than JTAG. Modern chips all use SWD but older chips may require JTAG.

Reply:
Fixed